openid/src/main/java/fish/payara/security/openid/domain/JsonClaims.java

```diff
@@ -79,14 +80,19 @@
             return Collections.emptyList();
         }
         if (value.getValueType() == JsonValue.ValueType.STRING) {
-            return Collections.singletonList(value.toString());
+            return Collections.singletonList(getStringValue(value));
         }
         if (value.getValueType() == JsonValue.ValueType.ARRAY) {
-            return value.asJsonArray().stream().map(JsonValue::toString).collect(Collectors.toList());
+            return value.asJsonArray().stream().map(this::getStringValue).collect(Collectors.toList());
         }
         throw new IllegalArgumentException("Cannot interpret "+name+" as string array");
     }
 
+    private String getStringValue(JsonValue value) {
+            JsonString stringValue = (JsonString)value;
```

Comment:
@pdudits, should this method throw an exception if value is not a string? Or should non-string values be just ignored? For example, the method getArrayStringClaim throws IllegalArgumentException instead of returning an empty list. Is it better to terminate a request with an exception? Or just return an empty value and continue?

Reply:
My proposal would to be as friendly as possible.
So let's have getStringValue do something like:

```
switch (value.getValueType) {
   case JsonValue.TRUE:
      return "true";
   case JsonValue.FALSE:
     return "false";
   case JsonValue.STRING:
     return ((JsonString)value).getString();
   case JsonValue.NUMBER:
     return ((JsonNumber)value).numberValue().toString();
   default:
    throw new IllegalArgumentException("Cannot handle nested JSON value in a claim:" + value);
  }
}
```

We can extend the if at line 82 to handle everything except `ValueType.ARRAY`.

Reply:
I added what you suggested and also changed getArrayStringClaim() to accept arrays and anything that the new getStringValue method accepts.
I also added some debug logging just in case users need to ivnestigate what their provider sends and how Payara handles it.